    void \u02ce(byte[]) throws java.io.IOException
    {
        java.io.OutputStream $r2;
        java.lang.Throwable $r3, $r4;
        byte[] $r1;
        util.h.xy.ct.mg r0;
        java.lang.NullPointerException r5;
        byte $b2;
        int $i0, $i1;
        boolean $z0;

        r0 := @this: util.h.xy.ct.mg;

        $r1 := @parameter0: byte[];

        $i0 = <util.h.xy.ct.mg: int \u02bb>;

        $i1 = $i0 ^ 113;

        $i0 = $i0 & 113;

        $i0 = $i0 << 1;

        $i1 = $i1 + $i0;

        $i0 = $i1 % 128;

        <util.h.xy.ct.mg: int \u037a> = $i0;

        $i1 = $i1 % 2;

        if $i1 == 0 goto label01;

        $z0 = 1;

        goto label02;

     label01:
        $z0 = 0;

     label02:
        if $z0 == 1 goto label03;

        $r2 = r0.<util.h.xy.ct.mg: java.io.OutputStream \u02cf>;

        virtualinvoke $r2.<java.io.OutputStream: void write(byte[])>($r1);

        goto label05;

     label03:
        $r2 = r0.<util.h.xy.ct.mg: java.io.OutputStream \u02cf>;

        virtualinvoke $r2.<java.io.OutputStream: void write(byte[])>($r1);

     label04:
        r5 = new java.lang.NullPointerException;

        specialinvoke r5.<java.lang.NullPointerException: void <init>(java.lang.String)>("Invalid array reference replaced by Soot");

        throw r5;

     label05:
        $i0 = <util.h.xy.ct.mg: int \u02bb>;

        $i1 = $i0 & 1;

        $i0 = $i0 | 1;

        $i1 = $i1 + $i0;

        $i0 = $i1 % 128;

        <util.h.xy.ct.mg: int \u037a> = $i0;

        $i1 = $i1 % 2;

        if $i1 == 0 goto label06;

        $b2 = 87;

        goto label07;

     label06:
        $b2 = 19;

     label07:
        if $b2 == 19 goto label11;

     label08:
        r5 = new java.lang.NullPointerException;

        specialinvoke r5.<java.lang.NullPointerException: void <init>(java.lang.String)>("This statement would have triggered an Exception: specialinvoke $u1#10.<java.lang.Object: int hashCode()>()");

        throw r5;

     label09:
        $r3 := @caughtexception;

        throw $r3;

     label10:
        $r4 := @caughtexception;

        throw $r4;

     label11:
        return;

        catch java.lang.Throwable from label04 to label05 with label10;
        catch java.lang.Throwable from label08 to label09 with label09;
    }
